apps/accounts/views.py

```diff
@@ -72,29 +99,59 @@ def login_user(request, user, identity):
     login(request, user)
 
 
+def render_error(request, error, next_path=None):
+    if resolve(request.path).func.func_name == 'AuthenticateView':
```

Comment:
This is fragile because you're relying on some Django implementation details. More importantly though, this logic feels like it's in the wrong place. The logic is specific to the AuthenticateView and would be totally hidden to someone staring at the AuthenticateView trying to understand its flow.

Instead, I would suggest changing the decorator a bit, to this:

````
def with_user(fn=None, error_handler=immediate_error_response):
    ...
````
This way, you can call it from `AuthenticateView` with a custom error handler:
````
class AuthenticateView(APIView):
    @with_user(error_handler=fxa_error_redirector)
    def get(self, ...):
         ...
````
This requires another layer of decorator nesting but you can [steal this code](https://github.com/mozilla/olympia/blob/master/apps/amo/decorators.py#L125-L139) to get that working.